outlines/generate/generator.py

```diff
@@ -125,6 +126,29 @@ def token_generator(model, sampler: "Sampler") -> Callable:
 
     """
 
+    def cache_key_args_for_generate(
+        token_ids: torch.Tensor,
+        attention_masks: torch.Tensor,
+        kv_cache: torch.Tensor,
+        allowed_tokens: List[List[int]],
+        rng: torch.Generator,
+    ):
+        """The function generates a tuple of arguments used as a key for caching the 'generate' function.
+
+        The cache key is based on the input token IDs, attention masks, KV cache, allowed token sequences,
+        random number generator seed, model configuration, and the name of the sampler function.
+        """
+        return (
+            token_ids,
+            attention_masks,
+            kv_cache,
+            allowed_tokens,
```

Comment:
@rlouf Want to draw your attention here, I am not hashing the lists by converting it to set here and rely on FSM to output the tokens in a sort of idempotent manner. 

Can you verify if my understanding is correct?